mobile/lib/common/loading_screen.dart

```diff
@@ -14,58 +17,100 @@ import 'package:go_router/go_router.dart';
 class LoadingScreen extends StatefulWidget {
   static const route = "/loading";
 
-  const LoadingScreen({super.key});
+  final Future<void>? restore;
+
+  const LoadingScreen({super.key, this.restore});
 
   @override
   State<LoadingScreen> createState() => _LoadingScreenState();
 }
 
 class _LoadingScreenState extends State<LoadingScreen> {
+  String message = "Welcome to 10101";
+
   @override
   void initState() {
-    super.initState();
-
-    Future.wait<dynamic>([
+    List<Future<dynamic>> futures = [
       Preferences.instance.hasEmailAddress(),
       Preferences.instance.getOpenPosition(),
       isSeedFilePresent(),
-    ]).then((value) {
+      Preferences.instance.isFullBackupRequired(),
+    ];
+
+    if (widget.restore != null) {
+      // wait for the restore process to finish!
+      futures.add(widget.restore!);
+    }
+
+    Future.wait<dynamic>(futures).then((value) {
       final hasEmailAddress = value[0];
       final position = value[1];
       final isSeedFilePresent = value[2];
+      final isFullBackupRequired = value[3];
 
-      logger.d("Scanning for seed file: $isSeedFilePresent");
+      FlutterNativeSplash.remove();
 
       if (isSeedFilePresent) {
-        runBackend(context).then((value) {
-          logger.i("Backend started");
-
-          if (!hasEmailAddress) {
-            GoRouter.of(context).go(WelcomeScreen.route);
+        if (!hasEmailAddress) {
+          GoRouter.of(context).go(WelcomeScreen.route);
+        } else {
+          if (isFullBackupRequired) {
+            setState(() => message = "Creating initial backup!");
+            fullBackup().then((value) {
```

Comment:
There is no exception handling `.catch` on this future.

Reply:
f8d3b7b065fdf97be524d42eea54efcc4a5066c7